src/solver/conv_hip_implicit_gemm_grouped_wrw_xdlops.cpp

```diff
@@ -250,22 +308,63 @@ static std::vector<std::string> GetKernelAsTokens(const std::string& kernel)
     return tokens;
 }
 
-void PerformanceConfigHipImplicitGemmGroupWrwXdlops::InitHeuristicKernelIDs()
+void PerformanceConfigHipImplicitGemmGroupWrwXdlops::InitHeuristicKernelIDs(const std::string& type)
 {
     for(int i = 0; i < valid_kernels.size(); i++)
     {
-        if(valid_kernels[i].find("DeviceGroupedConvBwdWeight_Xdl_CShuffle") != std::string::npos)
+        if(valid_kernels[i].find(type) != std::string::npos)
         {
             heuristic_indexes.push_back(i);
-            heuristic_kernels.push_back(GetKernelAsTokens(valid_kernels[i]));
+            heuristic_kernels[i] = GetKernelAsTokens(valid_kernels[i]);
         }
     }
 }
 
-bool PerformanceConfigHipImplicitGemmGroupWrwXdlops::ModelApplyToken(int idx, std::string value)
+bool PerformanceConfigHipImplicitGemmGroupWrwXdlops::ModelApplyToken(
+    int idx, std::string value, const std::string& arch, const ProblemDescription& problem)
 {
-    if(idx == 13)
+    if(idx == 13 && arch == "gfx90a")
         idx += 1; // skip
+    if(arch == "gfx942")
+    {
+        if(idx == 0)
+        {
+            InitHeuristicKernelIDs(value);
+            if(!valid_kernels.empty())
+                return true;
+            return false;
+        }
+        if(idx > 0)
+            idx--;
+        if(idx >= 12)
+            idx += 2;
+        if(((idx == 15 && (valid_kernels[heuristic_indexes[0]].find(
+                               "DeviceGroupedConvBwdWeight_Xdl_CShuffle") != std::string::npos)) ||
```

Comment:
if `CK` decides to change the name of their kernels, we have to come find all the locations where we hardcoded it.

Reply:
I'll have it check the size of the kernels within valid_kernel instead.